### PR TITLE
[flutter_tools] rename output LICENSE file to NOTICES and support loading either

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Work towards https://github.com/flutter/flutter/issues/16723

This is only safe to land after https://github.com/flutter/flutter/pull/58131 lands in google3. Only build NOTICES in asset manfiest, and load either LICENSE or NOTICES from pubspec dependencies.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
FYI @chingjun are there any places in google3 that are using  `LICENSE` as output by the current flutter tool? Or is that already handled by blaze

**Comment by chingjun** [comment]
We do not depend on the `LICENSE` file output by flutter_tools.

We do, however, generate a `LICENCE` file and place it in the assets directory in the app bundle. Does it need to be changed to `NOTICES`?

**Comment by jonahwilliams** [comment]
Yeah, it would need to eventually change to NOTICES

 I could update the framework logic to be non-breaking by: check for a NOTICES and then if that is missing check for LICENSES

**Comment by jonahwilliams** [comment]
I opened https://github.com/flutter/flutter/pull/58131 to allow us to opt into notices in google3 before I land this

**Comment by jonahwilliams** [comment]
With #58131  patched in this is non-breaking for google3

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/asset.dart [hunk 1]
```diff
@@ -395,28 +395,23 @@ List<_Asset> _getMaterialAssets(String fontSet) {
...
+/// which the component license applies, with each package name on its own line
+/// and the list of package names separated from the actual license text by a
+/// blank line. The packages need not match the names of the pub package. For
 /// example, a package might itself contain code from multiple third-party
-/// sources, and might need to include a license for each one.)
-// Note: this logic currently has a bug, in that we collect LICENSE information
-// for dev_dependencies and transitive dev_dependencies. These are not actually
-// compiled into the released application and don't need to be included. Unfortunately,
-// the pubspec.yaml alone does not have enough information to determine which
-// dependencies are transitive of dev_dependencies, so a simple filter isn't sufficient.
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/asset.dart:419]: was this note wrong or did it get fixed?

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/asset.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/asset.dart:419]: It was wrong, per our discussion a while back :)

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/asset.dart:419]: I guess we talked about it like, months ago, but seems like it was last week!

**Pull request merged**